File: app/src/main/java/org/scp/gymlog/ui/training/rows/TrainingBitRow.java
package org.scp.gymlog.ui.training.rows;

import org.scp.gymlog.model.Bit;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter @AllArgsConstructor
public class TrainingBitRow implements ITrainingRow {
    private final Bit bit;

    @Override
    public Type getType() {
        return Type.BIT;
    }
}

File: app/src/main/java/org/scp/gymlog/ui/training/rows/TrainingBitRow.kt
package org.scp.gymlog.ui.training.rows

import org.scp.gymlog.model.Bit

class TrainingBitRow(
    val bit: Bit
) : ITrainingBitRow {
    override val type = if (bit.superSet > 0)
        ITrainingBitRow.Type.BIT_SUPERSET
    else
        ITrainingBitRow.Type.BIT
}